docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md

```diff
@@ -0,0 +1,251 @@
+    return Fraction(top, bottom);
+}
+
+Fraction operator -(const Fraction f1, const int f2){
+    int top = f1.numerator - f2*f1.denomenator;
+    int bottom = f1.denomenator;
+    return Fraction(top, bottom);
+}
+
+Fraction operator -(const int f1, const Fraction f2){
+    int top = f1*f2.denomenator - f2.numerator;
+    int bottom = f2.denomenator;
+    return Fraction(top, bottom);
+}
+
+Fraction operator *(const Fraction f1, const int f2){
+    int top = f1.numerator*f2;
+    int bottom = f1.denomenator;
+    return Fraction(top, bottom);
+}
+
+Fraction operator *(const int f1, const Fraction f2){
+    int top = f1*f2.numerator;
+    int bottom = f2.denomenator;
+    return Fraction(top, bottom);
+}
+
+Fraction operator /(const Fraction f1, const int f2){
+    int top = f1.numerator;
+    int bottom = f1.denomenator*f2;
+    return Fraction(top, bottom);
+}
+
+Fraction operator /(const int f1, const Fraction f2){
+    int top = f2.numerator;
+    int bottom = f2.denomenator*f1;
+    return Fraction(top, bottom);
+}
+
+bool operator ==(const Fraction f1, const Fraction f2){
+    int result = f1.numerator*f2.denomenator - f2.numerator*f1.denomenator;
+    if(result == 0){
+        return true;
+    }
+    else{
+        return false;
+    }
+}
+
+bool operator >(const Fraction f1, const Fraction f2){
+    int result = f1.numerator*f2.denomenator - f2.numerator*f1.denomenator;
+    if(result > 0){
+        return true;
+    }
+    else{
+        return false;
+    }
+}
+
+bool operator <(const Fraction f1, const Fraction f2){
+    int result = f1.numerator*f2.denomenator - f2.numerator*f1.denomenator;
+    if(result < 0){
+        return true;
+    }
+    else{
+        return false;
+    }
+}
+
+bool operator >=(const Fraction f1, const Fraction f2){
+    return !(f1 < f2);
+}
+
+bool operator <=(const Fraction f1, const Fraction f2){
+    return !(f1 > f2);
+}
+
+bool operator !=(const Fraction f1, const Fraction f2){
+    return !(f1 == f2);
+}
+
+int main(){
+    // Testing above class methods
+    Fraction f1(6, 2);
+    Fraction f2(1, 4);
+    cout<<-f1<<endl;
+    f1 += f2;
+    cout<<f1+f2<<endl;
+    cout<<f1*f2<<endl;
+    cout<<f1-f2<<endl;
+    cout<<f1/f2<<endl;
+    cout<<(f1 <= f2)<<endl;
+    return 0;
+}
+```
+
+Here, we defined the `Fraction` class which we'll use to provide new data type for working with fractions.
+Inside this class, we begin by creating two private attributes numerator and denomenator.
+Constructor of fractions takes two optional arguments top and bottom where top will be assigned to numerator and
+bottom will be assigned to denometor.Here we checks if denometor is zeros then we throws an error and if denomenator is
+negative then we negate top and bottom before assigning them.
+We also have to implement getters for this Class and a method for unary operator `-`.
+We also have to implement friend functions(one can use normal method) for binary operators like +, -, >=, == etc.
```

Comment:
Here, we try to include a section for how to run the solution. It appears your solution is more of a library and less of an executable program.